server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -338,4 +339,17 @@ public int hashCode() {
             return Objects.hashCode(key) ^ Objects.hashCode(value);
         }
     }
+
+    /**
+     * Converts the supplied list to a hash map where each key is the list's index.
```

Comment:
I think it is better if we not specify exact type in javadoc as long as actual return type is `Map`. Also would it be better to return immutable map here?

Reply:
Sure, will remove "hash" from the javadoc.

We cannot use an immutable map, because one part where the returned Map is used is during the "builder" which gradually adds stuff to it. But I can check whether there other places where an immutable map could be used, and if so, will make another version of this.

Reply:
Finally the place where the map is used is mutable.